plugins/scaffolder/src/components/TemplateEditorPage/DryRunResults/DryRunResultsList.tsx

```diff
@@ -67,9 +68,18 @@ export function DryRunResultsList() {
             </ListItemIcon>
             <ListItemText primary={`Result ${result.id}`} />
             <ListItemSecondaryAction>
+              <IconButton
+                edge="end"
+                aria-label="download"
+                title="Download as .zip"
+                onClick={() => dryRun.downloadResult(result.id)}
```

Comment:
I'd say it's best to inline the downloading logic into this component, or potentially split it out to a separate helper. On top of that we can track the state of the action and keep the download button disabled while it's in progress.

Reply:
done